### ISSUE TITLE
create-space without quota does not use default

### ISSUE THREAD

**Issue by Zteve** [cloudfoundry/cli#774]
### Command
cf create-space new-space
### CLI Version
cf version 6.15.0+fa1bfe2-2016-01-13
### Error
No default quota is set on the space.
### Stack Trace
N/A
### CC API Version
230, but N/A
### CF Trace
...includes...
```
POST /v2/spaces?async=true&inline-relations-depth=1 HTTP/1.1
{"name":"new-space","organization_guid":"df6486e2-e31c-4223-80d1-d04e6fb493d1"}
```
...fuller trace below...
### Platform Details
OS X El Capitan Version 10.11.3
### Shell
Terminal

The help text talks about a "default quota". This doesn't appear to be obtained from anywhere.

**Comment by Zteve** [comment]
This is a (slightly edited) cf trace of the command `cf create-space new-space`:
```
Creating space new-space in org integration-test as admin...

REQUEST: [2016-02-23T13:13:20Z]
POST /v2/spaces?async=true&inline-relations-depth=1 HTTP/1.1

{"name":"new-space","organization_guid":"df6486e2-e31c-4223-80d1-d04e6fb493d1"}

RESPONSE: [2016-02-23T13:13:20Z]
HTTP/1.1 201 Created

{
  "metadata": {
    "guid": "77b549cd-263d-4c38-a912-fa331d4c2bbb",
    "url": "/v2/spaces/77b549cd-263d-4c38-a912-fa331d4c2bbb",
    "created_at": "2016-02-23T13:13:20Z",
    "updated_at": null
  },
  "entity": {
    "name": "new-space",
    "organization_guid": "df6486e2-e31c-4223-80d1-d04e6fb493d1",
    "space_quota_definition_guid": null,
    "allow_ssh": true,
    "organization_url": "/v2/organizations/df6486e2-e31c-4223-80d1-d04e6fb493d1",
    "organization": {
      "metadata": {
        "guid": "df6486e2-e31c-4223-80d1-d04e6fb493d1",
        "url": "/v2/organizations/df6486e2-e31c-4223-80d1-d04e6fb493d1",
        "created_at": "2016-02-23T09:26:29Z",
        "updated_at": null
      },
      "entity": {
        "name": "integration-test",
        "billing_enabled": false,
        "quota_definition_guid": "e8705d39-0084-47f6-9ce1-7e08b79f8cf4",
        "status": "active",
        "quota_definition_url": "/v2/quota_definitions/e8705d39-0084-47f6-9ce1-7e08b79f8cf4",
        "spaces_url": "/v2/organizations/df6486e2-e31c-4223-80d1-d04e6fb493d1/spaces",
        "domains_url": "/v2/organizations/df6486e2-e31c-4223-80d1-d04e6fb493d1/domains",
        "private_domains_url": "/v2/organizations/df6486e2-e31c-4223-80d1-d04e6fb493d1/private_domains",
        "users_url": "/v2/organizations/df6486e2-e31c-4223-80d1-d04e6fb493d1/users",
        "managers_url": "/v2/organizations/df6486e2-e31c-4223-80d1-d04e6fb493d1/managers",
        "billing_managers_url": "/v2/organizations/df6486e2-e31c-4223-80d1-d04e6fb493d1/billing_managers",
        "auditors_url": "/v2/organizations/df6486e2-e31c-4223-80d1-d04e6fb493d1/auditors",
        "app_events_url": "/v2/organizations/df6486e2-e31c-4223-80d1-d04e6fb493d1/app_events",
        "space_quota_definitions_url": "/v2/organizations/df6486e2-e31c-4223-80d1-d04e6fb493d1/space_quota_definitions"
      }
    },
    "developers_url": "/v2/spaces/77b549cd-263d-4c38-a912-fa331d4c2bbb/developers",
    "developers": [

    ],
    "managers_url": "/v2/spaces/77b549cd-263d-4c38-a912-fa331d4c2bbb/managers",
    "managers": [

    ],
    "auditors_url": "/v2/spaces/77b549cd-263d-4c38-a912-fa331d4c2bbb/auditors",
    "auditors": [

    ],
    "apps_url": "/v2/spaces/77b549cd-263d-4c38-a912-fa331d4c2bbb/apps",
    "apps": [

    ],
    "routes_url": "/v2/spaces/77b549cd-263d-4c38-a912-fa331d4c2bbb/routes",
    "routes": [

    ],
    "domains_url": "/v2/spaces/77b549cd-263d-4c38-a912-fa331d4c2bbb/domains",
    "domains": [
      {
        "metadata": {
          "guid": "10e30fe3-82e8-4967-9838-54a4462bd097",
          "url": "/v2/domains/10e30fe3-82e8-4967-9838-54a4462bd097",
          "created_at": "2016-02-23T09:24:06Z",
          "updated_at": "2016-02-23T09:24:06Z"
        },
        "entity": {
          "name": "local.micropcf.io",
          "router_group_guid": null
        }
      }
    ],
    "service_instances_url": "/v2/spaces/77b549cd-263d-4c38-a912-fa331d4c2bbb/service_instances",
    "service_instances": [

    ],
    "app_events_url": "/v2/spaces/77b549cd-263d-4c38-a912-fa331d4c2bbb/app_events",
    "events_url": "/v2/spaces/77b549cd-263d-4c38-a912-fa331d4c2bbb/events",
    "security_groups_url": "/v2/spaces/77b549cd-263d-4c38-a912-fa331d4c2bbb/security_groups",
    "security_groups": [
      {
        "metadata": {
          "guid": "91331bc5-b973-4b3b-bc90-7d756a6e54c6",
          "url": "/v2/security_groups/91331bc5-b973-4b3b-bc90-7d756a6e54c6",
          "created_at": "2016-02-23T09:24:06Z",
          "updated_at": null
        },
        "entity": {
          "name": "public_networks",
          "rules": [
            {
              "destination": "0.0.0.0-9.255.255.255",
              "protocol": "all"
            },
            {
              "destination": "11.0.0.0-169.253.255.255",
              "protocol": "all"
            },
            {
              "destination": "169.255.0.0-172.15.255.255",
              "protocol": "all"
            },
            {
              "destination": "172.32.0.0-192.167.255.255",
              "protocol": "all"
            },
            {
              "destination": "192.169.0.0-255.255.255.255",
              "protocol": "all"
            }
          ],
          "running_default": true,
          "staging_default": true,
          "spaces_url": "/v2/security_groups/91331bc5-b973-4b3b-bc90-7d756a6e54c6/spaces"
        }
      },
      {
        "metadata": {
          "guid": "ca90f4f0-7d82-4013-ab09-7365236c7dd1",
          "url": "/v2/security_groups/ca90f4f0-7d82-4013-ab09-7365236c7dd1",
          "created_at": "2016-02-23T09:24:06Z",
          "updated_at": null
        },
        "entity": {
          "name": "dns",
          "rules": [
            {
              "destination": "0.0.0.0/0",
              "ports": "53",
              "protocol": "tcp"
            },
            {
              "destination": "0.0.0.0/0",
              "ports": "53",
              "protocol": "udp"
            }
          ],
          "running_default": true,
          "staging_default": true,
          "spaces_url": "/v2/security_groups/ca90f4f0-7d82-4013-ab09-7365236c7dd1/spaces"
        }
      }
    ]
  }
}
OK
Assigning role SpaceManager to user admin in org integration-test / space new-space as admin...

REQUEST: [2016-02-23T13:13:20Z]
PUT /v2/organizations/df6486e2-e31c-4223-80d1-d04e6fb493d1/users/50c3719b-90d1-4af3-85ff-400500444c9b HTTP/1.1

RESPONSE: [2016-02-23T13:13:20Z]
HTTP/1.1 201 Created

{
  "metadata": {
    "guid": "df6486e2-e31c-4223-80d1-d04e6fb493d1",
    "url": "/v2/organizations/df6486e2-e31c-4223-80d1-d04e6fb493d1",
    "created_at": "2016-02-23T09:26:29Z",
    "updated_at": null
  },
  "entity": {
    "name": "integration-test",
    "billing_enabled": false,
    "quota_definition_guid": "e8705d39-0084-47f6-9ce1-7e08b79f8cf4",
    "status": "active",
    "quota_definition_url": "/v2/quota_definitions/e8705d39-0084-47f6-9ce1-7e08b79f8cf4",
    "spaces_url": "/v2/organizations/df6486e2-e31c-4223-80d1-d04e6fb493d1/spaces",
    "domains_url": "/v2/organizations/df6486e2-e31c-4223-80d1-d04e6fb493d1/domains",
    "private_domains_url": "/v2/organizations/df6486e2-e31c-4223-80d1-d04e6fb493d1/private_domains",
    "users_url": "/v2/organizations/df6486e2-e31c-4223-80d1-d04e6fb493d1/users",
    "managers_url": "/v2/organizations/df6486e2-e31c-4223-80d1-d04e6fb493d1/managers",
    "billing_managers_url": "/v2/organizations/df6486e2-e31c-4223-80d1-d04e6fb493d1/billing_managers",
    "auditors_url": "/v2/organizations/df6486e2-e31c-4223-80d1-d04e6fb493d1/auditors",
    "app_events_url": "/v2/organizations/df6486e2-e31c-4223-80d1-d04e6fb493d1/app_events",
    "space_quota_definitions_url": "/v2/organizations/df6486e2-e31c-4223-80d1-d04e6fb493d1/space_quota_definitions"
  }
}

REQUEST: [2016-02-23T13:13:20Z]
PUT /v2/spaces/77b549cd-263d-4c38-a912-fa331d4c2bbb/managers/50c3719b-90d1-4af3-85ff-400500444c9b HTTP/1.1

RESPONSE: [2016-02-23T13:13:20Z]
HTTP/1.1 201 Created

{
  "metadata": {
    "guid": "77b549cd-263d-4c38-a912-fa331d4c2bbb",
    "url": "/v2/spaces/77b549cd-263d-4c38-a912-fa331d4c2bbb",
    "created_at": "2016-02-23T13:13:20Z",
    "updated_at": null
  },
  "entity": {
    "name": "new-space",
    "organization_guid": "df6486e2-e31c-4223-80d1-d04e6fb493d1",
    "space_quota_definition_guid": null,
    "allow_ssh": true,
    "organization_url": "/v2/organizations/df6486e2-e31c-4223-80d1-d04e6fb493d1",
    "developers_url": "/v2/spaces/77b549cd-263d-4c38-a912-fa331d4c2bbb/developers",
    "managers_url": "/v2/spaces/77b549cd-263d-4c38-a912-fa331d4c2bbb/managers",
    "auditors_url": "/v2/spaces/77b549cd-263d-4c38-a912-fa331d4c2bbb/auditors",
    "apps_url": "/v2/spaces/77b549cd-263d-4c38-a912-fa331d4c2bbb/apps",
    "routes_url": "/v2/spaces/77b549cd-263d-4c38-a912-fa331d4c2bbb/routes",
    "domains_url": "/v2/spaces/77b549cd-263d-4c38-a912-fa331d4c2bbb/domains",
    "service_instances_url": "/v2/spaces/77b549cd-263d-4c38-a912-fa331d4c2bbb/service_instances",
    "app_events_url": "/v2/spaces/77b549cd-263d-4c38-a912-fa331d4c2bbb/app_events",
    "events_url": "/v2/spaces/77b549cd-263d-4c38-a912-fa331d4c2bbb/events",
    "security_groups_url": "/v2/spaces/77b549cd-263d-4c38-a912-fa331d4c2bbb/security_groups"
  }
}
OK
Assigning role SpaceDeveloper to user admin in org integration-test / space new-space as admin...

REQUEST: [2016-02-23T13:13:20Z]
PUT /v2/organizations/df6486e2-e31c-4223-80d1-d04e6fb493d1/users/50c3719b-90d1-4af3-85ff-400500444c9b HTTP/1.1

RESPONSE: [2016-02-23T13:13:20Z]
HTTP/1.1 201 Created

{
  "metadata": {
    "guid": "df6486e2-e31c-4223-80d1-d04e6fb493d1",
    "url": "/v2/organizations/df6486e2-e31c-4223-80d1-d04e6fb493d1",
    "created_at": "2016-02-23T09:26:29Z",
    "updated_at": null
  },
  "entity": {
    "name": "integration-test",
    "billing_enabled": false,
    "quota_definition_guid": "e8705d39-0084-47f6-9ce1-7e08b79f8cf4",
    "status": "active",
    "quota_definition_url": "/v2/quota_definitions/e8705d39-0084-47f6-9ce1-7e08b79f8cf4",
    "spaces_url": "/v2/organizations/df6486e2-e31c-4223-80d1-d04e6fb493d1/spaces",
    "domains_url": "/v2/organizations/df6486e2-e31c-4223-80d1-d04e6fb493d1/domains",
    "private_domains_url": "/v2/organizations/df6486e2-e31c-4223-80d1-d04e6fb493d1/private_domains",
    "users_url": "/v2/organizations/df6486e2-e31c-4223-80d1-d04e6fb493d1/users",
    "managers_url": "/v2/organizations/df6486e2-e31c-4223-80d1-d04e6fb493d1/managers",
    "billing_managers_url": "/v2/organizations/df6486e2-e31c-4223-80d1-d04e6fb493d1/billing_managers",
    "auditors_url": "/v2/organizations/df6486e2-e31c-4223-80d1-d04e6fb493d1/auditors",
    "app_events_url": "/v2/organizations/df6486e2-e31c-4223-80d1-d04e6fb493d1/app_events",
    "space_quota_definitions_url": "/v2/organizations/df6486e2-e31c-4223-80d1-d04e6fb493d1/space_quota_definitions"
  }
}

REQUEST: [2016-02-23T13:13:20Z]
PUT /v2/spaces/77b549cd-263d-4c38-a912-fa331d4c2bbb/developers/50c3719b-90d1-4af3-85ff-400500444c9b HTTP/1.1

RESPONSE: [2016-02-23T13:13:20Z]
HTTP/1.1 201 Created

{
  "metadata": {
    "guid": "77b549cd-263d-4c38-a912-fa331d4c2bbb",
    "url": "/v2/spaces/77b549cd-263d-4c38-a912-fa331d4c2bbb",
    "created_at": "2016-02-23T13:13:20Z",
    "updated_at": null
  },
  "entity": {
    "name": "new-space",
    "organization_guid": "df6486e2-e31c-4223-80d1-d04e6fb493d1",
    "space_quota_definition_guid": null,
    "allow_ssh": true,
    "organization_url": "/v2/organizations/df6486e2-e31c-4223-80d1-d04e6fb493d1",
    "developers_url": "/v2/spaces/77b549cd-263d-4c38-a912-fa331d4c2bbb/developers",
    "managers_url": "/v2/spaces/77b549cd-263d-4c38-a912-fa331d4c2bbb/managers",
    "auditors_url": "/v2/spaces/77b549cd-263d-4c38-a912-fa331d4c2bbb/auditors",
    "apps_url": "/v2/spaces/77b549cd-263d-4c38-a912-fa331d4c2bbb/apps",
    "routes_url": "/v2/spaces/77b549cd-263d-4c38-a912-fa331d4c2bbb/routes",
    "domains_url": "/v2/spaces/77b549cd-263d-4c38-a912-fa331d4c2bbb/domains",
    "service_instances_url": "/v2/spaces/77b549cd-263d-4c38-a912-fa331d4c2bbb/service_instances",
    "app_events_url": "/v2/spaces/77b549cd-263d-4c38-a912-fa331d4c2bbb/app_events",
    "events_url": "/v2/spaces/77b549cd-263d-4c38-a912-fa331d4c2bbb/events",
    "security_groups_url": "/v2/spaces/77b549cd-263d-4c38-a912-fa331d4c2bbb/security_groups"
  }
}
OK

TIP: Use 'cf target -o "integration-test" -s "new-space"' to target new space
```

**Comment by dkoper** [comment]
Hi @Zteve,

Thanks for reporting this issue!
Looks like the option's help was copied from `create-org` - there is no default for spaces.
I'll ask the team to take a look.

Cheers,
Dies Koper
CF CLI PM

**Comment by dkoper** [comment]
This has been fixed in cf CLI v6.17.0.
https://github.com/cloudfoundry/cli#downloads

Thanks again!
Dies Koper
CF CLI PM
